src/components/editor/Bindings/SchemaInference/Button.tsx

```diff
@@ -56,32 +56,10 @@ function SchemaInferenceButton() {
             )
                 .then(
                     (response) => {
-                        let inferredSpec = null;
-                        if (Object.hasOwn(collectionData.spec, 'writeSchema')) {
-                            const { ...additionalSpecKeys } =
-                                collectionData.spec;
-
-                            inferredSpec = !isEmpty(response.schema)
-                                ? {
-                                      ...additionalSpecKeys,
-                                      writeSchema:
-                                          collectionData.spec.writeSchema,
-                                      readSchema: response.schema,
-                                  }
-                                : null;
-                        } else {
-                            // Removing schema from the object
-                            const { schema, ...additionalSpecKeys } =
-                                collectionData.spec;
-
-                            inferredSpec = !isEmpty(response.schema)
-                                ? {
-                                      ...additionalSpecKeys,
-                                      writeSchema: collectionData.spec.schema,
-                                      readSchema: response.schema,
-                                  }
-                                : null;
-                        }
+                        const inferredSpec = moveUpdatedSchemaToReadSchema(
+                            collectionData,
+                            response.schema
+                        );
```

Comment:
Initially I thought we'd need this function with the editor but that got removed. However, keeping this functionality in a shared spot feels right still.